### ISSUE TITLE
Improve Accessibility Labeling

### ISSUE THREAD

**Issue by kspearrin** [bitwarden/mobile#187]
There are a lot of accessibility labels missing on buttons and inputs that make screen readers more effective.

**Comment by Flameborn** [comment]
The two most needed parts that require accessible labels are the tabs on the bottom, as well as the two buttons next to the entries list.

In addition, the app becomes inaccessible after logging in, and the only fix is to restart the application, after which VoiceOver is able to detect the elements on-screen.

**Comment by steverep** [comment]
I would add that many actionable items are not given a proper accessible role, such as a button or link.  For example, the "Regenerate Password" button on the generator tab is just announced by VoiceOver as normal text and oddly says "empty list" before it.

In context, it's usually clear what is clickable, but this should be fixed for better accessibility as well.

**Comment by pitermach** [comment]
Updating this issue for Bitwarden 2.1.0.51 (iOS) not sure about the state of accessibility on Android.

The bottom 3 tabs now have labels, as do the search and add buttons at the top.
However, on the main view/edit item screen, most buttons have lost their labels. Edit, options, save and cancel buttons still have labels, but the controls to copy the username and password, as well as showing/hiding it and checking whether it was exposed are all now unlabeled and just read as "button". Same goes for the buttons next to each entry in the vault that display the overflow menu.

**Comment by kspearrin** [comment]
Thanks @pitermach . For the next version, I have added a11y labels to all the icon text and icon buttons that I could find.

**Comment by pitermach** [comment]
Hello Kyle,
Awesome, that’s great to hear. Thank you.

**Comment by djsmith85** [comment]
Closing this as per Kyle's comment, the a11y labels were added. If someone is still encountering missing accessibility labels, feel free to open a new issue.
